pkg/machine/ocipull/ociartifact.go

```diff
@@ -153,13 +167,46 @@ func (o *OCIArtifactDisk) get() error {
 		//
 		// i.e. 91d1e51...d28974.qcow2.xz
 		if err := o.unpack(artifactDigest); err != nil {
-			return err
+			return nil, err
 		}
 	} else {
 		logrus.Debugf("cached image exists and is latest: %s", cachedImagePath.GetPath())
 		o.cachedCompressedDiskPath = cachedImagePath
 	}
-	return nil
+	return cleanCache, nil
+}
+
+func (o *OCIArtifactDisk) cleanCache(cachedImagePath string) func() {
+	// cache miss while using an image that we cache, ie the default image
+	// clean out all old files fron the cache dir
+	if o.cache {
+		files, err := os.ReadDir(o.dirs.ImageCacheDir.GetPath())
+		if err != nil {
+			logrus.Warn("failed to clean machine image cache", err)
```

Comment:
this should use `Warn("text: ")` or `Warnf("text: %v", err)`, without this you add the erro without any space, same in other places.